addons/product/models/product.py

```diff
@@ -795,6 +804,14 @@ def _find_suitable_product_packaging(self, product_qty, uom_id):
                 return packaging
         return self.env['product.packaging']
 
+    @api.constrains('barcode')
```

Comment:
The check method should be bellow the SQL constraint as in the guidelines. It's correct atm for this class so let keep it clean.